### PR TITLE
Added login process

### PR DESCRIPTION
**Pull request by purple-void**
-Added login, logout, and register actions
-Paths for endpoints are in urls
-Functionality are in views
-Register form is in forms
-html files are currently in templates folder in backend (can be changed later) -Token for register is in token

### CODE REVIEW

File: backend/core/constants.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import os
+
+EMAIL_FROM = os.getenv("EMAIL_FROM", "")
```
Comment (seporterfield) [inline backend/core/constants.py:3]: What does constants.py do differently than settings.py?

Comment (seporterfield) [reply backend/core/constants.py]: What does django do with this variable/ how does it find it?

Comment (purple-void) [reply backend/core/constants.py]: it looks cooler

File: backend/rebutify/urls.py [hunk 2]
```diff
@@ -15,20 +15,15 @@
...
 from django.urls import include, path
-from rest_framework import routers
-
-router = routers.DefaultRouter()
-router.register(r"status/alive", views.StatusViewSet, basename="alive")
-router.register(r"users", views.UserViewSet)
-router.register(r"groups", views.GroupViewSet)
 
 urlpatterns = [
-    path(os.getenv("DJANGO_ADMIN_PATH", "admin/"), admin.site.urls),
```
Comment (seporterfield) [inline backend/rebutify/urls.py:31]: Why delete all the DRF paths?

File: backend/requirements.txt [hunk 3]
```diff
@@ -2,4 +2,6 @@ django==5.0.4
 python-dotenv==1.0.1
 gunicorn==22.0.0
 uvicorn==0.29.0
-djangorestframework==3.15.1
\ No newline at end of file
+djangorestframework==3.15.1
+crispy-bootstrap4==2024.1
+django-crispy-forms==2.1
```
Comment (seporterfield) [inline backend/requirements.txt:7]: maintenance on this library doesn't look good, let's look for alternatives

**Review by seporterfield — CHANGES_REQUESTED** [review]

File: backend/rebutify/urls.py (hunk 2 above)
Comment (seporterfield) [reply backend/rebutify/urls.py:31]: Waiting for localhost:8000/api/status/alive -> 200 before resolve

File: backend/requirements.txt (hunk 3 above)
Comment (seporterfield) [reply backend/requirements.txt:7]: Resolving since templating is gonna be temporary anyway, moving to next.js

**Review by seporterfield — APPROVED** [review]
After we finish integrating with next.js, we gotta
- write tests for the login flow paths
- replace the ALLOWED_HOST workaround
- remove unused templates and dependencies

Looks good!

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -14,14 +14,15 @@ repos:
...
 
+# commenting out temporarily for error
   # VALIDATE BRANCH NAME
-  - repo: local
-    hooks:
-      - id: validate-branch-name
-        name: Validating Branch Name
-        entry: git-hooks/validate-branch-name
-        language: script
-        stages: [pre-commit, post-checkout]
```
Comment (Monstarrrr) [inline .pre-commit-config.yaml:24]: Can remove the comment entirely

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -83,20 +84,21 @@ repos:
...
-        args:
-          [
-            '--tab-width',
-            '2',
-            '--single-quote',
-            '--no-semi',
-            '--jsx-single-quote',
-            '--print-width',
-            '68',
-          ]
```
Comment (Monstarrrr) [inline .pre-commit-config.yaml:102]: Can remove the comment entirely

File: backend/.template-env [hunk 6]
```diff
@@ -16,6 +16,9 @@ DJANGO_ADMIN_PATH=admin/
 # Hostname
 DJANGO_HOST=localhost
 
+# Extra host value for ALLOWED_HOSTS (TODO: replace)
```
Comment (Monstarrrr) [inline backend/.template-env:19]: Why is there a TODO?

File: backend/core/forms.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import User
+
+
+class UserRegisterForm(UserCreationForm):
+    email = forms.EmailField()
+
+    class Meta:
+        model = User
+        fields = ["username", "email", "password1", "password2"]
```
Comment (Monstarrrr) [inline backend/core/forms.py:11]: Those variable names aren't explicit enough plus they use numbers which I personally think should be avoided.  
password1 is "password", password2 is "confirmPassword"

**Review by Monstarrrr — CHANGES_REQUESTED** [review]

File: backend/.template-env (hunk 6 above)
Comment (seporterfield) [reply backend/.template-env:19]: this PR is about getting email login to work
Would it be more appropriate to keep this workaround and come back to this in a different PR or better to refactor it here and now?

Comment (seporterfield) [reply backend/.template-env]: fixed

**Review by seporterfield — APPROVED** [review]

File: backend/core/forms.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import User
+
+
+class UserRegisterForm(UserCreationForm):
+    email = forms.EmailField()
+
+    class Meta:
+        model = User
+        fields = ["username", "email"]
```
Comment (Monstarrrr) [inline backend/core/forms.py:11]: No password & confirmPassword fields ?

Comment (Monstarrrr) [reply backend/core/forms.py:11]: I see it added here though https://github.com/Monstarrrr/rebutify/pull/31/commits/c9337e8c43aed4d818a62f649606ff648a5642a1 , I don't understand.

Comment (Monstarrrr) [reply backend/core/forms.py:11]: I see you removed them here https://github.com/Monstarrrr/rebutify/pull/31/commits/2410fa92594e027b533db045a367cacbb0a26f4b

Comment (seporterfield) [reply backend/core/forms.py:11]: those fields never existed in the creation form in the first place, so I removed them

**Review by Monstarrrr — APPROVED** [review]

**Pull request merged**